python/letsql/backends/let/__init__.py

```diff
@@ -290,8 +293,38 @@ def _get_source(self, expr: ir.Expr):
     def _register_and_transform_cache_tables(self, expr):
         """This function will sequentially execute any cache node that is not already cached"""
 
+        tables = {}
+        count = itertools.count()
+
         def fn(node, _, **kwargs):
+            for k, v in list(kwargs.items()):
+                try:
+                    if v in tables:
+                        name = f"{v.name}_{next(count)}"
+                        kwargs[k] = DatabaseTable(
+                            name,
+                            schema=v.schema,
+                            source=v.source,
+                            namespace=v.namespace,
+                        )
+                        remote: RemoteTable = tables[v]
+                        batches = remote.remote_expr.to_pyarrow_batches()
```

Comment:
I think this doesn't address the case where there are multiple passes on the table within the single engine expr.